### DISCUSSION TITLE
python API on Unity

### DISCUSSION THREAD

**Discussion by dana4056** [microsoft/AirSim#3460] [Support Q&A] [answered]
Unity: 2019.3.12
python: 3.8.5

I want to use Python API in Unity.
I cloned AirSim and connected Unity with Python.
However, I can't move car through API.

I modified the setting.json file, 
and I also modified the cpp and cs file as https://github.com/microsoft/AirSim/pull/2937 instructed, but I can only move the car with the keyboard.

Please help me move the car through API.

in Unity!!

**Comment by rajat2004** [comment, marked as answer]
I just tested #2937 again, and it seems to be working (at least partially). Did you rebuild AirSim after applying the changes? Maybe run `clean.sh` and rebuild again

**Reply by dana4056** [reply]
Thanks a lot. After modifying the file, I rebuilt the airsim and rebuilt Unity, and it was resolved. 
I haven't solved it for weeks. you're an angel!!!!!!!!!!

**Reply by dana4056** [reply]
Resolution Process (in Windows 10)
1. Modify cpp and cs file (https://github.com/microsoft/AirSim/pull/2937)
2. Rebuild AirSim
     2.1.  `cd AirSim`
     2.2.  `clean_rebuild`
3. Unity Rebuild
     3.1. ` cd AirSim\Unity`
     3.2.  `build.cmd`
